From d0384a7caed036d25228fa3279c36c0230795e4a Mon Sep 17 00:00:00 2001
From: Willow Barraco <contact@willowbarraco.fr>
Date: Fri, 11 Aug 2023 08:49:37 +0200
Subject: [PATCH sxmo-utils] Switch from light to brightnessctl

light has been flagged as orphan by its developper, and dropped from
aports.

https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/49859
https://github.com/haikarainen/light/issues/147

We switch to brightnessctl that works the same, and that also support
flashlight and leds.

Signed-off-by: Willow Barraco <contact@willowbarraco.fr>
---
 scripts/core/sxmo_brightness.sh | 23 ++++++++++++++++-------
 1 file changed, 16 insertions(+), 7 deletions(-)

diff --git a/scripts/core/sxmo_brightness.sh b/scripts/core/sxmo_brightness.sh
index aec00020..5bbeed7a 100755
--- a/scripts/core/sxmo_brightness.sh
+++ b/scripts/core/sxmo_brightness.sh
@@ -10,27 +10,36 @@
 
 notify() {
 	if [ "$SXMO_WM" = "sway" ] && [ -z "$SXMO_WOB_DISABLE" ]; then
-		light | grep -o "^[0-9]*" > "$XDG_RUNTIME_DIR"/sxmo.wobsock
+		getvalue > "$XDG_RUNTIME_DIR"/sxmo.wobsock
 	else
-		light | xargs dunstify -r 888 "$icon_brightness Brightness"
+		getvalue | xargs dunstify -r 888 "$icon_brightness Brightness"
 	fi
 }
 
 setvalue() {
-	light -S "$1"
+	brightnessctl set "$1"%
 }
 
 up() {
-	light -A 5
+	brightnessctl set 5%+
 }
 
 down() {
-	light -N "${SXMO_MIN_BRIGHTNESS:-5}"
-	light -U 5
+	# bugged https://github.com/Hummer12007/brightnessctl/issues/82
+	# brightnessctl --min-value "${SXMO_MIN_BRIGHTNESS:-5}" set 5%-
+
+	if [ "$(getvalue)" -gt "${SXMO_MIN_BRIGHTNESS:-5}" ]; then
+		brightnessctl set 5%-
+	else
+		brightnessctl set "${SXMO_MIN_BRIGHTNESS:-5}"%
+	fi
 }
 
 getvalue() {
-	light
+	brightnessctl info \
+		| grep "Current brightness:" \
+		| awk '{ print $NF }' \
+		| grep -o "[0-9]*"
 }
 
 "$@"
-- 
2.41.0

